omnibus/files/private-chef-cookbooks/private-chef/templates/default/nginx/nginx_chef_api_lb.conf.erb

```diff
@@ -55,6 +55,15 @@
       proxy_pass <%= node['private_chef']['data_collector']['root_url'] %>;
     }
     <% end -%>
+
+    <% if node['private_chef']['data_collector']['root_url'] && node['private_chef']['data_collector']['token'] -%>
+    location ~ /organizations/([^/]+)/data-collector {
+      set $request_org $1;
+      access_by_lua_file '<%= @script_path %>/check_token_using_endpoint.lua';
+      proxy_pass $upstream;
```

Comment:
This is a neat idea: having the lua script decide which upstream is the right one. :+1:

However, since it's currently hard-coded to data-collector-ish stuff only anyways, can't we put [that logic](https://github.com/chef/chef-server/pull/984/files/bb0efc646b250eac89d39eb7010265d7b7dbc5e0#diff-9376b5038260ef7c52869f5018a6feceR18) here?